ci/unit_tests/functions_test/main_test.py

```diff
@@ -0,0 +1,268 @@
+        for missingTestArgIndex, missingTestArg in enumerate(self.functionsTestArgs[::2]):
+            print(missingTestArg)
+            print(missingTestArgIndex)
+            providedTestArgs = list(self.functionsTestArgs)
+            del providedTestArgs[2 * missingTestArgIndex + 1]
+            del providedTestArgs[2 * missingTestArgIndex]
+
+            providedTestArgs = ['/some/random/path', '/some/random/path'] + providedTestArgs
+            if missingTestArg in ['--cloudfunctions_package', '--cloudfunctions_function']:
+                self.t_exitCodeAndLogMessage(
+                    1, # exit code
+                    'WARNING  \'' + missingTestArg[2:] + '\' parameter not defined', # warning message substring
+                    [providedTestArgs] # params (*args, **kwargs)
+                )
+            else:
+                self.t_exitCodeAndLogMessage(
+                    1, # exit code
+                    'ERROR    required \'' + missingTestArg[2:] + '\' parameter not defined', # error message substring
+                    [providedTestArgs] # params (*args, **kwargs)
+                )
+
+    def test_nonExistentFileInput(self):
+        ''' Tests if the input file does not exist '''
+        testArgs = ['/some/random/path', '/some/random/path'] + self.functionsTestArgs
+        self.t_exitCodeAndLogMessage(
+            1, # exit code
+            'CRITICAL Cannot open test input file /some/random/path', # critical message substring
+            [testArgs] # params (*args, **kwargs)
+        )
+
+    def test_nonExistentFileOutput(self):
+        ''' Tests if the output file does not exist '''
+        testArgs = [self.noJsonPath, '/some/random/path'] + self.functionsTestArgs
+        self.t_exitCodeAndLogMessage(
+            1, # exit code
+            'CRITICAL Cannot open test output file /some/random/path', # critical message substring
+            [testArgs] # params (*args, **kwargs)
+        )
+
+    def test_invalidJsonInput(self):
+        ''' Tests if the input file contains invalid json '''
+        testArgs = [self.noJsonPath, self.noJsonPath] + self.functionsTestArgs
+        self.t_exitCodeAndLogMessage(
+            1, # exit code
+            'CRITICAL Cannot decode json from test input file ' + 
+            '/home/tomas/Projects/watson-assistant-workbench/ci/unit_tests/functions_test/main_data/no.json, ' + 
+            'error: No JSON object could be decoded', # error message substring
+            [testArgs] # params (*args, **kwargs)
+        )
+
+    def test_nonArrayJsonInput(self):
+        ''' Tests if the input file contains non array json '''
+        testArgs = [self.emptyDictJsonPath, self.noJsonPath] + self.functionsTestArgs
+        self.t_exitCodeAndLogMessage(
+            1, # exit code
+            'CRITICAL Input test json is not array!', # error message substring
+            [testArgs] # params (*args, **kwargs)
+        )
+
+    def test_testSingleInvalid(self):
```

Comment:
I think that name `test_singleInvalid` would be sufficient (same for tests bellow).

Reply:
The second 'test' in the name means that in the input test file there is only one test and not more.